lib/Models/Cesium.ts

```diff
@@ -428,6 +410,59 @@ export default class Cesium extends GlobeOrMap {
     });
   }
 
+  /** Add an event listener to a TerrainProvider.
+   * If we get an error when trying to load the terrain, then switch to smooth mode, and notify the user.
+   * Finally, remove the listener, so failed tiles do not trigger the error as these can be common and are not a problem. */
+  private async catchTerrainProviderDown(terrainProvider: TerrainProvider) {
+    // Some network errors are not rejected through readyPromise, so we have to
+    // listen to them using the error event and dispose it later
+    let networkErrorListener: (err: any) => void;
+    const networkErrorPromise = new Promise((_resolve, reject) => {
+      networkErrorListener = reject;
+      terrainProvider.errorEvent.addEventListener(networkErrorListener);
+    });
+
+    const isReady = await Promise.race([
+      networkErrorPromise,
+      terrainProvider.readyPromise
+    ])
+      .then(() => {
+        /** Need to throw an error if incorrect `cesiumTerrainUrl` has been specified.
+        The terrainProvider.readyPromise will still be fulfulled, but the map will not load correctly
+        So we check for terrainProvider.availability */
+        if (!terrainProvider.availability) {
+          throw new Error();
+        }
+      })
+      .catch((err) => {
+        console.log("Terrain provider error.  ", err.message);
+        if (this.scene.terrainProvider instanceof CesiumTerrainProvider) {
+          console.log("Switching to EllipsoidTerrainProvider.");
+          setViewerMode("3dsmooth", this.terriaViewer);
+          if (!this._terrainMessageViewed) {
+            this.terria.raiseErrorToUser(err, {
+              title: "Terrain Server Not Responding",
+              message:
+                "\
+The terrain server is not responding at the moment.  You can still use all the features of " +
+                this.terria.appName +
+                " \
+but there will be no terrain detail in 3D mode.  We're sorry for the inconvenience.  Please try \
+again later and the terrain server should be responding as expected.  If the issue persists, please contact \
+us via email at " +
+                this.terria.supportEmail +
+                "."
```

Comment:
Might be good to move these strings to the translation file? 

Example:
https://github.com/TerriaJS/terriajs/blob/8a051a7845c6625c53bc50e76920060ccdf22d35/lib/Models/Cesium.ts#L583-L588

Reply:
Done